src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -172,65 +172,83 @@ namespace nupic {
               sp2.getMinPctActiveDutyCycles()));
 
 
-    Real boostFactors1[numColumns];
-    Real boostFactors2[numColumns];
+    auto boostFactors1 = new Real[numColumns];
```

Comment:
I'm failing to find previous discussion on this, but this is a GCC extension (C99). Designated initializers didn't make it into C++11, so this change occurred to cope with compilers that lacked this extension. Not sure whether to leave like this, or #ifdef a non-GCC version?